aten/src/ATen/native/native_functions.yaml

```diff
@@ -1466,8 +1468,12 @@
 
 - func: scalar_tensor(Scalar s, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
 
+- func: rand(int[] size, *, Dimname[]? names, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
```

Comment:
I think everything after the `*` is optional and kwarg only, is there a reason we can't just provide a default None value and merge this with the existing interface below?

Reply:
Yes, there is a reason and it is complicated. The short answer is that this is the easiest way to implement it.

Long answer: We've got two rand declarations in native_functions.yaml (without the Generator arg):
```
- func: rand(int[] size, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
- func: rand(int[] size, *, Tensor(a!) out) -> Tensor(a!)
```
These get merged together to produce one scheme string for the python arg parser:
```
    "rand(IntArrayRef size, *, Tensor out=None, ScalarType dtype=None, Layout layout=torch.strided, Device device=None, bool pin_memory=False, bool requires_grad=False)",
```
Now, my claim is that it doesn't make sense to add a `names=` argument when an `out` tensor is provided. If there were such an arg, it could either 
- (1) check that the `out` tensor has the correct `names`. In this case, the user would always have to specify the correct names which is bad because it's verbose and unnecessary.
- (2) override the `names` of the `out` tensor: this is weird, especially if `names=None` by default.

Because it doesn't make sense, `rand(int[] size, *, Dimname[]? names, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor`
should be a separate schema that isn't combined with an `out=` argument.

Reply:
ah weird, OK

Reply:
It's weird and I don't like it because it means we need 2x the schemas and that it's easy for the schemas to go out of sync. But I don't know of how else we can do this.